### PR TITLE
Purchased albums photos

### PR DESCRIPTION
**Pull request by manueltuero**
## Summary
I added a new endpoint to allow get photos of purchased albums by album id

## Trello Card
https://trello.com/c/7mboJr8H

### CODE REVIEW

File: app/controllers/albums.js [hunk 1]
```diff
@@ -28,3 +30,29 @@ exports.getAlbumsByUser = (req, res, next) =>
   Album.findAlbumsByUser(req)
     .then(albums => res.status(200).send(albums))
     .catch(next);
+
+exports.getPurchasedAlbumPhotos = async (req, res, next) => {
+  try {
+    const albumId = req.params.id;
+    let response = {};
+    let query = {};
```
Comment (jersonlopez) [inline app/controllers/albums.js]: `let response = query = {};`

File: test/utils.js [hunk 2]
```diff
@@ -39,3 +39,28 @@ exports.mockPurchaseAlbum = () => {
       title: 'quidem molestiae enim'
     });
 };
+
+exports.mockAlbumNotPurchasedError = id => {
+  nock(url)
+    .get(`/users/albums/${id}/photos`)
```
Comment (jersonlopez) [inline test/utils.js]: this endpoint is our it should not be mocked

File: test/albums.test.js [hunk 3]
```diff
@@ -196,4 +205,133 @@ describe('albums api tests', () => {
...
+      body: { token }
+    } = await request(server)
+      .post('/users/sessions')
+      .query({
+        email: user2.email,
+        password: 'Wolox1189!'
+      });
+
+    mockPurchaseAlbum();
+    mockAlbumNotPurchasedError(1);
```
Comment (jersonlopez) [inline test/albums.test.js]: same here!

**Review by jersonlopez — CHANGES_REQUESTED** [review]

File: app/controllers/albums.js (hunk 1 above)
Comment (manueltuero) [reply app/controllers/albums.js]: This not works with eslint actual config :c

File: app/controllers/albums.js [hunk 4]
```diff
@@ -28,3 +30,22 @@ exports.getAlbumsByUser = (req, res, next) =>
...
+  try {
+    const query = { albumId: req.params.id };
+    if (!req.decoded.admin) {
+      query.userId = req.decoded.id;
+    }
+    const album = await Album.findAlbum(query);
+    if (album) {
+      logger.info(`Finding the photos of album ${album.albumId} ...`);
+      const photos = await getAll(`photos?albumId=${query.albumId}`);
+      return res.status(200).send(photos);
```
Comment (jersonlopez) [inline app/controllers/albums.js]: return is unnecessary

File: app/controllers/albums.js [hunk 5]
```diff
@@ -28,3 +30,22 @@ exports.getAlbumsByUser = (req, res, next) =>
...
+    }
+    logger.error(`The album id ${query.albumId} photos could not be obtained`);
+    return next(notFoundError(`The album id ${query.albumId} photos could not be obtained`));
```
Comment (jersonlopez) [inline app/controllers/albums.js:47]: same here!

File: app/controllers/albums.js [hunk 6]
```diff
@@ -28,3 +30,22 @@ exports.getAlbumsByUser = (req, res, next) =>
...
+  } catch (err) {
+    return next(err);
```
Comment (jersonlopez) [inline app/controllers/albums.js:49]: and here!

**Review by jersonlopez — CHANGES_REQUESTED** [review]

**Review by jersonlopez — APPROVED** [review]

**Pull request merged**